### PR TITLE
Added new metric application_buildpack

### PR DESCRIPTION
**Pull request by adamspd**

### CODE REVIEW

**Comment by benjaminguttmann-avtq** [comment]
@adamspd just to ask one upfront question, are all those commits related to the new application_buildpack metric? If not it would be appreciated if they could spread across different PRs which would make it way easier to review them

**Comment by adamspd** [comment]
Hi @benjaminguttmann-avtq, the commits were not related to a single metric but to the addition of 2 new metrics and the modification of a metric to add more information. I separated them into 3 pull requests as requested and yes, you're right, it will be easier later if we need to go back on one of them.

File: fetcher/fetcher.go [hunk 1]
```diff
@@ -94,10 +94,15 @@ func (c *Fetcher) fetch() *models.CFObjects {
 
 	c.workInit()
 
+	// Fetch and filter droplets
+	appDropletMap, err := c.fetchAndFilterDroplets(session)
```
Comment (gmllt) [inline fetcher/fetcher.go:98]: this should be in the list of fetchers executed by the workers.

the fetch function should be added to the fetchers pool using `c.worker.PushIf("droplets", c.fetchDroplets, filters.Droplets)`

**Review by gmllt — CHANGES_REQUESTED** [review]
the fetchAndFilterDroplets() should be added to the workers execution pool using `c.worker.PushIf("droplets", c.fetchDroplets, filters.Droplets)`

File: collectors/applications.go [hunk 2]
```diff
@@ -242,6 +258,18 @@ func (c ApplicationsCollector) reportApp(application models.Application, objs *m
 		buildpack = appSum.DetectedBuildpack
 	}
 
+	// 3. Use the droplet data for the buildpack metric
+	apps, exists := objs.Apps[application.GUID]
```
Comment (gmllt) [inline collectors/applications.go:262]: is there a specific reason to get the application from the obj.Apps slice here ?

**Review by gmllt — CHANGES_REQUESTED** [review]

File: collectors/applications.go (hunk 2 above)
Comment (adamspd) [reply collectors/applications.go]: I had the droplet logic in mind when initially reviewing the code, which is why I included that line. However, considering the current implementation, you're right. The line of code fetching the application from `objs.Apps` is unnecessary and can be removed. Thank you for pointing that out.

**Comment by benjaminguttmann-avtq** [comment]
@adamspd Please make sure to run `go test -v ./...` as well as `golangci-lint run` to see if there are any issues

**Comment by adamspd** [comment]
> @adamspd Please make sure to run `go test -v ./...` as well as `golangci-lint run` to see if there are any issues

Absolutely.

File: fetcher/fetcher.go [hunk 3]
```diff
@@ -81,6 +81,7 @@ func (c *Fetcher) workInit() {
 	c.worker.PushIf("service_bindings", c.fetchServiceBindings, filters.ServiceBindings)
 	c.worker.PushIf("users", c.fetchUsers, filters.Events)
 	c.worker.PushIf("events", c.fetchEvents, filters.Events)
+	c.worker.PushIf("application_buildpack", c.fetchApplications, filters.Applications)
```
Comment (gmllt) [inline fetcher/fetcher.go:84]: not necessary, fetchApplications has already been pushed to workers at line 63

**Review by gmllt — CHANGES_REQUESTED** [review]

File: fetcher/fetcher.go (hunk 3 above)
Comment (adamspd) [reply fetcher/fetcher.go]: Fixed

**Review by gmllt — APPROVED** [review]
LGTM thanks @adamspd

**Pull request merged**